### ISSUE TITLE
pathGroup "toBeParsed" not working as expected.

### ISSUE THREAD

**Issue by PaRoxUs** [kangax/fabric.js#2418]
What you see here is 3 different paths printed 1 time with a strokeWidth on 14 and color #ccc and one time each with a strokeWidth 12 and color #fff, then one of them is printed with strokeWidth 1 and color #000.

How this behaves with SVG is that the line would be centered above each other, leaving a effect of a on 1px around the stroke that is printed with strokeWidth 12 and color #fff, but as you can see the paths is top and left aligned, now is this just how the path stroke rendering is set up or is this a bug?

![skarmavbild 2015-08-13 kl 13 12 14](https://cloud.githubusercontent.com/assets/1275755/9248553/fb46bf28-41bc-11e5-980a-47981be3ea11.png)

```
var tmpl = {
    shadowCollectiveAbstractMidle:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#cccccc", strokeWidth:14, fill:false }
        },
        shadowCollectiveLogicMidle:{
            path:'M410.7,859.1v-126l-44-62v-404l44-32v-50',
            set:{ stroke:"#cccccc", strokeWidth:14, fill:false }
        },
        shadowIndividualMidle:{
            path:'M388.7,185.1v650',
            set:{ stroke:"#cccccc", strokeWidth:14, fill:false }
        },
        collectiveAbstractMidle:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        collectiveAbstractMidle:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        collectiveAbstractMidle2:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#000", strokeWidth:1, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        collectiveLogicMidle:{
            path:'M410.7,859.1v-126l-44-62v-404l44-32v-50',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        individualMidle:{
            path:'M388.7,185.1v650',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        }
};
var group = group || new fabric.Group([], { left: 0, top: 0 });

for (var name in tmpl) {
    var path = new fabric.Path(tmpl[name].path);

     path.set(tmpl[name].set);
    group.add(path);
}
```

**Comment by asturur** [comment]
If you want a SVG similar rendering, use pathGroup instead of Group.
```
// clear canvas
canvas.clear();

var tmpl = {
    shadowCollectiveAbstractMidle:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#cccccc", strokeWidth:14, fill:false }
        },
        shadowCollectiveLogicMidle:{
            path:'M410.7,859.1v-126l-44-62v-404l44-32v-50',
            set:{ stroke:"#cccccc", strokeWidth:14, fill:false }
        },
        shadowIndividualMidle:{
            path:'M388.7,185.1v650',
            set:{ stroke:"#cccccc", strokeWidth:14, fill:false }
        },
        collectiveAbstractMidle:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        collectiveAbstractMidle:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        collectiveAbstractMidle2:{
            path:'M366.7,185.1v50l44,32v404l-44,62v126',
            set:{ stroke:"#000", strokeWidth:1, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        collectiveLogicMidle:{
            path:'M410.7,859.1v-126l-44-62v-404l44-32v-50',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        },
        individualMidle:{
            path:'M388.7,185.1v650',
            set:{ stroke:"#fff", strokeWidth:12, fill:false, strokeLineCap:'round',strokeLinejoin:'round' }
        }
};

var paths = [];
for (var name in tmpl) {
    var path = new fabric.Path(tmpl[name].path);
    path.set(tmpl[name].set);
    paths.push(path);
}
var group = new fabric.PathGroup(paths, { left: 0, top: 0 });
group.setCoords();
canvas.add(group);
```

![image](https://cloud.githubusercontent.com/assets/1194048/9248879/1697adac-41c0-11e5-9a14-860c1a8b54b0.png)

Is not about stroke behaviour, is about path positioning.

**Comment by PaRoxUs** [comment]
Thank you very much! Is there any particular reason why Controls are not shown for a PathGroup when using Canvas?

**Comment by asturur** [comment]
Because we are not creating it with standard svg parsing.

You can do:

```
var group = new fabric.PathGroup(paths, {top:0, left:0, toBeParsed: true });
```

But something is polluting the coordinates including a 0,0 that is enlarging the bounding box.
this is unexpected.

**Comment by PaRoxUs** [comment]
This might be a similar issue.

The canvas is set to 1024x1024p and the the Centered "BodyGraph" is rendered through loadSVGFromURL is positioned in the center of the SVG file. The other "BodyGraph" that is below and not centered is rendered with the fabric.Path function after me copying the paths from the same SVG file. What I expected was that the paths copied from the SVG would render centered too. Would not that be  the expected behavior? 

![skarmavbild 2015-08-14 kl 10 54 53](https://cloud.githubusercontent.com/assets/1275755/9270547/eff1634c-4272-11e5-91f6-1e1cfe0ccebc.png)

**Comment by asturur** [comment]
Did you notice that if you paste some code, i can help you faster?
Really is not possible to help by guessing what are you doing.

**Comment by PaRoxUs** [comment]
Here you go man!
```		
	fabric.loadSVGFromURL('/images/BodyGraph-centered.svg', function(objects, options) {
		var obj = fabric.util.groupSVGElements(objects, options);
		canvas.add(obj);
		canvas.renderAll();
	});

	var d = "M319.7,163.1c18-28,56-30,72-30s40.077,0,66.077,30c37.922,44,11.924,90-4.076,118c-16,30-16,54-4,70c20,24,42,28,69.998,32 c14,2,20,8,24,18c13.438,32.479,102.002,240,128.002,306c3.998,10,3.998,20,3.998,20c0,6-4,14-8,18 c-96,108-163.998,138-275.999,138c-112,0-179.999-30-276-138c-4-4-8.076-12-8-18c0,0,0-10,4.077-20 c25.923-66,104.312-251.084,127.923-306c4-10,10-16,24-18c28-4,50-8,70-32c8-10,11.858-27.838,10.076-32 c-2.271-5.875-6.457-6.151-18.077-4c-6,0-16-2-16-12s0-32-2-34c-2-4-10.078-4-10-10c0-2,14-28,14-32c0-4-4-10-6-12 	S305.7,185.1,319.7,163.1z"
	var path = new fabric.Path(d);
	path.set({ stroke: '#737373', strokeWidth:2 });
	canvas.add(path);
	canvas.renderAll();
```

**Comment by PaRoxUs** [comment]
Do you need the SVG too?

**Comment by asturur** [comment]
The problem is that the svg start from coordinates x = 0, Y = 0, and the path is probably centered by coordinates. It means that it is drawn from x = 319.7, y = 163.1 and going on, it appears on the center.
If you modifiy width and height of SVG file, probably is not in the center anymore.

Probably in the SVG file, in a group containing the paths, there is some transform attribute with translate componente, or there is a viewbox that starts with negative coordiantes.

Please post SVG.

**Comment by asturur** [comment]
You have this viewBox:
-123.3 -12.9 ... ....

you have to do path.left -= -123.3 and path.top -= -12.9

But is an hack, you may encounter strange situation in which parsing the svg is the only way.

**Comment by PaRoxUs** [comment]
Thank you! I missed this because the Art board in AI is set to 0, 0 but after some infestation I found out that there is an issue with Illustrator and viewBox when you export to SVG although there is an solution that hopefully can help someone else too:
http://graphicdesign.stackexchange.com/questions/15401/change-viewbox-attribute-in-svg-exported-by-illustrator.

**Comment by PaRoxUs** [comment]
Here is another example where the bounding box is polluted. 
![skarmavbild 2015-08-28 kl 09 57 48](https://cloud.githubusercontent.com/assets/1275755/9541738/5215c93e-4d6b-11e5-97cf-f867b923048a.png)

here is the code:
```
var path = 'M440,176c18-28,56-30,72-30c16,0,40.077,0,66.077,30c37.923,44,11.924,90-4.076,118c-16,30-16,54-4,70c20,24,42,28,69.999,32 c14,2,20,8,24,18c13.438,32.479,102.001,240,128.001,306C796,730,796,740,796,740c0,6-4,14-8,18c-96,108-163.999,138-276,138	c-112,0-179.999-30-276-138c-4-4-8.076-12-8-18c0,0,0-10,4.077-20C258,654,336.389,468.916,360,414c4-10,10-16,24-18	c28-4,50-8,70-32c8-10,11.858-27.838,10.076-32c-2.271-5.875-6.457-6.151-18.077-4c-6,0-16-2-16-12s0-32-2-34c-2-4-10.078-4-10-10	c0-2,14-28,14-32c0-4-4-10-6-12S426,198,440,176z'

var path = new fabric.Path(path);
path.set({ fill:"#000" });

var group = new fabric.PathGroup([path], { toBeParsed:true });

```

**Comment by asturur** [comment]
Is not really polluted.
You need some behaving between groups and pathgroups, that actually we do not have.

**Comment by PaRoxUs** [comment]
I think it is weird that the bounding box of PathGroups behaves like this and this also effects the width of the group! What I expect is a bounding box just around the paths.

**Comment by PaRoxUs** [comment]
at least there should be a trimBoundingBox option for this.

**Comment by asturur** [comment]
We take a slow down to add options for every use case.
"toBeParsed" is an option used internally to fix svgs that come with "width" and "height" undeclared in the header.
We need to fix groups (not pathgroups) to behave correctly and this take a little bit of time and effort.
Adding options to make patches is not a good idea. It make documentation complex, user confused.

**Comment by PaRoxUs** [comment]
Thanks for the clarification, I understand!

**Comment by dud3** [comment]
Just hit the same issue.

**Comment by asturur** [comment]
i think this can be closed. This kind of defects have been mostly solved in 2.0 branch
